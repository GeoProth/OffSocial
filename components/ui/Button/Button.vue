<template>
  <button class="ui-button" @click.prevent="clickAction">
    {{ text }}
  </button>
</template>
<script>
export default {
  name: "Button",
  props: {
    text: [String, Number],
    clickAction: {
      type: Function,
      default: () => null
    }
  }
};
</script>
<style lang="stylus" scoped>
  .ui-button {
    display: inline-block
    text-align: center
    vertical-align: middle
    border-radius: 0.5rem
    height: 2rem
    width: 5.5rem
    border: none
  }
  .ui-button:hover {
    cursor: pointer
  }
  .ui-button:active {
    outline: none
    border: none
  }
  .ui-button:focus { outline: 0 }
</style>
