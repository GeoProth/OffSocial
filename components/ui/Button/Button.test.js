import Button from "./Button";

describe("UI Button", () => {
  let wrapper;
  const clickAction = jest.fn();
  beforeEach(() => {
    wrapper = mountMock(Button, {
      props: {
        clickAction
      }
    });
  });
  test("the component can render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("the button will fire the click action when clicked", () => {
    const button = wrapper.find("button");
    button.trigger("click");
    expect(clickAction).toBeCalled();
  });
  test("clickAction prop defaults to a null function", () => {
    wrapper = mount(Button);
    expect(wrapper.props("clickAction")).toBeFunction();
    expect(wrapper.props("clickAction")()).toBe(null);
  });
});
