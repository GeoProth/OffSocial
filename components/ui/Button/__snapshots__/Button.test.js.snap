// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Button the component can render 1`] = `
<button class="ui-button">

</button>
`;
